src/components/tableV2/hooks/useInfiniteScroll.js

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useMemo, useState } from "react"
+
+const useIniniteScroll = ({
+  service,
+  target,
+  onError,
+  onSuccess,
+  onDataend,
+  updateTableData,
+  threshold = 100,
+}) => {
+  const [lastReceivedData, setLastReceivedData] = useState()
+  const [loading, setLoading] = useState(false)
+
+  const hasDataEnded = useMemo(() => {
+    if (!onDataend) return false
+    return onDataend(lastReceivedData)
+  }, [lastReceivedData])
+
+  const runService = async () => {
+    try {
+      const currentData = await service(lastReceivedData)
```

Comment:
This is in order to get the page/cursor payload? In any case `lastReceivedData` should be passed as a dependency to the `useEffect`

Reply:
You are right, i made a fix that introduces a new hook `useScrollListener` this will accept the `target` and `event handler` as input. The event handler will be saved as `ref` to not trigger a rerender.